File: PasswordValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;

namespace BreachBox
{
    internal class PasswordValidator
    {
        public static bool checkKey(String input) {
            // Ch3ckm@t3!
            String password = "Ch3ckm@t3!";
            return password.Equals(input);
        }

        public static bool secureCheckKey(String input)
        {
            // ByteMeUp!256
            string password = "OTgxMzI3ODnpVGlEssEGYvDjavBt0CXI64X+T7ag+VQxSsZgEHxTmw==";
            String hashedPassword = HashPassword(input, Encoding.UTF8.GetBytes("98132789"));
            return password.Equals(hashedPassword);
        }

        static string HashPassword(string password, byte[] salt)
        {
            using (var sha256 = new SHA256Managed())
            {
                byte[] passwordBytes = Encoding.UTF8.GetBytes(password);
                byte[] saltedPassword = new byte[passwordBytes.Length + salt.Length];

                // Concatenate password and salt
                Buffer.BlockCopy(passwordBytes, 0, saltedPassword, 0, passwordBytes.Length);
                Buffer.BlockCopy(salt, 0, saltedPassword, passwordBytes.Length, salt.Length);

                // Hash the concatenated password and salt
                byte[] hashedBytes = sha256.ComputeHash(saltedPassword);

                // Concatenate the salt and hashed password for storage
                byte[] hashedPasswordWithSalt = new byte[hashedBytes.Length + salt.Length];
                Buffer.BlockCopy(salt, 0, hashedPasswordWithSalt, 0, salt.Length);
                Buffer.BlockCopy(hashedBytes, 0, hashedPasswordWithSalt, salt.Length, hashedBytes.Length);

                return Convert.ToBase64String(hashedPasswordWithSalt);
            }
        }
    }
}

File: BreachBoxVB/PasswordValidator.vb
Imports System.Security.Cryptography

Public Class PasswordValidator
    Public Shared Function checkPassword(input As String) As Boolean
        ' Ch3ckm@t3
        Dim password = "Ch3ckm@t3"
        Return String.Compare(password, input) = 0
    End Function
    Public Shared Function checkPasswordSecure(input As String) As Boolean
        ' ByteMeUp!512
        Dim password = "d30442596ffa5dd3d286ce3e8130e20999f0749acc7c7d2800d3ab2f8601bb436c5cc35e3f6110295a8273e745a058d8a185943802752e410e071d02245cfcb7"
        Dim hashedPassword = HashPasswordSHA512(input)
        Return String.Compare(password, hashedPassword) = 0
    End Function


    'https://github.com/uk1337/VB.NET-Password-Hash-Functions/blob/main/HashFunctions.vb
    Private Shared Function HashPasswordSHA512(password As String) As String
        Dim sha512 = Security.Cryptography.SHA512.Create()
        Dim bytes As Byte() = System.Text.Encoding.UTF8.GetBytes(password)
        Dim hash As Byte() = sha512.ComputeHash(bytes)
        Dim stringBuilder As New System.Text.StringBuilder()
        For i As Integer = 0 To hash.Length - 1
            stringBuilder.Append(hash(i).ToString("x2"))
        Next
        Return stringBuilder.ToString()
    End Function

End Class
